File: Sem3/Advanced Methods of Programming/MyInterpreter - JavaFX - Final Project/project/src/domain/adt/MyStack.java
package domain.adt;


import java.util.ArrayList;
import java.util.List;
import java.util.Stack;


public class MyStack<TElement> implements MyStackInterface<TElement>
{
    private Stack<TElement> stack;


    public MyStack()
    {
        stack = new Stack<TElement>();
    }


    @Override
    public void push(TElement element)
    {
        stack.push(element);
    }


    @Override
    public TElement pop() throws AdtException
    {
        if (stack.empty())
            throw new AdtException("Pop from empty stack.");
        return stack.pop();
    }


    @Override
    public TElement top() throws AdtException
    {
        if (stack.empty())
            throw new AdtException("Top from empty stack.");
        return stack.peek();
    }


    @Override
    public boolean isEmpty()
    {
        return stack.empty();
    }


    @Override
    public String toString()
    {
        String str = "";
        Stack<TElement> stk = new Stack<TElement>();

        for (TElement element : stack)
        {
            stk.push(element);
        }

        for (TElement element : stk)
        {
            str += element.toString() + "\n";
        }
        return str;
    }


    @Override
    public List<TElement> getAll()
    {
        return new ArrayList<>(stack);
    }

}

File: Sem3/Advanced Methods of Programming/MyInterpreter - CSharp Console Application/MyInterpreter CSharp/domain/adt/MyStack.cs
using System.Collections.Generic;

namespace MyInterpreter_CSharp.domain.adt
{
    public class MyStack<TElement> : IMyStack<TElement>
    {
        private Stack<TElement> _stack;

        public MyStack()
        {
            _stack = new Stack<TElement>();
        }

        public void Push(TElement element)
        {
            _stack.Push(element);
        }

        public TElement Pop()
        {
            if (_stack.Count == 0)
                throw new AdtException("Pop from empty stack.");
            return _stack.Pop();
        }

        public bool IsEmpty()
        {
            return _stack.Count == 0;
        }

        public override string ToString()
        {
            string str = "";
            Stack<TElement> stk = new Stack<TElement>();

            foreach (TElement element in _stack)
            {
                stk.Push(element);
            }

            foreach (TElement element in stk)
            {
                str += element + "\n";
            }

            return str;
        }
    }
}